linera-views/src/map_view.rs

```diff
@@ -258,40 +310,53 @@ where
     ///   let mut map = ByteMapView::load(context).await.unwrap();
     ///   map.insert(vec![0,1], String::from("Hello"));
     ///   map.insert(vec![1,2], String::from("Bonjour"));
-    ///   map.insert(vec![2,2], String::from("Bonjour"));
+    ///   map.insert(vec![1,3], String::from("Bonjour"));
+    ///   let prefix = vec![1];
     ///   let mut count = 0;
     ///   map.for_each_key_while(|_key| {
     ///     count += 1;
-    ///     Ok(count < 1)
-    ///   }).await.unwrap();
-    ///   assert_eq!(count, 1);
+    ///     Ok(count < 3)
+    ///   }, prefix).await.unwrap();
+    ///   assert_eq!(count, 2);
     /// # })
     /// ```
-    pub async fn for_each_key_while<F>(&self, mut f: F) -> Result<(), ViewError>
+    pub async fn for_each_key_while<F>(&self, mut f: F, prefix: Vec<u8>) -> Result<(), ViewError>
     where
         F: FnMut(&[u8]) -> Result<bool, ViewError> + Send,
     {
-        let mut updates = self.updates.iter();
+        let prefix_len = prefix.len();
+        let mut updates = self.updates.range(get_interval(prefix.clone()));
         let mut update = updates.next();
```

Comment:
It would help the reader (or me, at least :sweat_smile:) to move these lines further down, just above "`if !self.was_cleared`", because I don't think these are used before that.

Reply:
Ok.